### PR TITLE
Implement Object.assign routine

### PR DESCRIPTION
**Pull request by rerobika**
Related part of the standard: ECMA-262 v6, 19.1.2.1

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-object.c [hunk 1]
```diff
@@ -922,6 +922,127 @@ ecma_builtin_object_object_define_property (ecma_value_t this_arg, /**< 'this' a
...
+        /* 5.c.iii.1 */
+        ecma_value_t prop_value = ecma_op_object_get (from_obj_p, property_name_p);
+
+        /* 5.c.iii.2 */
+        if (ECMA_IS_VALUE_ERROR (prop_value))
+        {
+          ret_value = prop_value;
+        }
+
+        if (ecma_is_value_empty (ret_value))
```
[the remark this answers is not in the document]
Comment (rerobika) [reply jerry-core/ecma/builtin-objects/ecma-builtin-object.c]: Thanks, that's a good idea!

**Comment by rerobika** [comment]
@LaszloLango I've applied your suggestion, also rebased the PR to the latest master.

**Comment by dxjia** [comment]
Hi, is it possible to use this API at native side?

**Comment by rerobika** [comment]
Hi @dxjia!

You can get this builtin routine from the API in the following way:

``` c
jerry_value_t global_object = jerry_get_global_object ();
jerry_value_t prop_name = jerry_create_string ((const jerry_char_t *) "Object");
jerry_value_t buitlin_object = jerry_get_property (global_object, prop_name);
jerry_release_value (global_object);
// check if `buitlin_object ` is error

prop_name = jerry_create_string ((const jerry_char_t *) "assign");
jerry_value_t object_assign_routine = jerry_get_property (buitlin_object , prop_name);
jerry_release_value (prop_name);
jerry_release_value (buitlin_object );

// check if `object_assign_routine ` is error

jerry_value_t result = jerry_call_function (object_assign_routine, <this_value>, <arguments>, arguments_length>);

jerry_release_value (object_assign_routine );

// Use `result` or check if it error
jerry_release_value (result);
```

I hope the following snippet helps to resolve your problem.

**Comment by dxjia** [comment]
@rerobika 
Thanks for your quick response, your suggestion is very helpful, and it works for me.
Thank you very much.

**Pull request merged**